editoast/editoast_schemas/src/primitives/duration.rs

```diff
@@ -65,6 +65,13 @@ pub enum PositiveDurationError {
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct PositiveDuration(ChronoDuration);
 
+#[cfg(feature = "testing")]
+impl PositiveDuration {
+    pub fn new(duration: ChronoDuration) -> Self {
+        Self(duration)
```

Comment:
`PositiveDuration` needs to keep an invariant: it must be positive. So introducing a new constructor that doesn't verify that property is a hole in the design. You might take inspiration from the implementation below of `TryFrom`, although we might not need a faillible function for tests: maybe a simple `assert!()` would be enough.